packages/visual-editor/src/editor/TranslatableRichTextField.tsx

```diff
@@ -22,51 +22,80 @@ export function TranslatableRichTextField<
     render: ({ onChange, value }) => {
       const { i18n } = useTranslation();
       const locale = i18n.language;
-      const resolvedValue = getDisplayValue(value, locale);
-      const fieldLabel = (label && pt(label)) + ` (${locale})`;
+      const resolvedValue = resolveTranslatableRichText(value, locale);
+      const fieldLabel = label ? `${pt(label)} (${locale})` : "";
+
+      const [pendingMessageId, setPendingMessageId] = React.useState<
+        string | undefined
+      >();
 
       const { sendToParent: openConstantValueEditor } = useSendMessageToParent(
         "constantValueEditorOpened",
         TARGET_ORIGINS
       );
 
-      const [pendingMessageId, setPendingMessageId] = React.useState<
-        string | undefined
-      >();
       useReceiveMessage(
         "constantValueEditorClosed",
         TARGET_ORIGINS,
         (_, payload) => {
           if (pendingMessageId && pendingMessageId === payload?.id) {
-            handleNewValue(payload.value);
+            // Handle the new Storm payload structure with locale, rtfJson, and rtfHtml
+            if (payload.locale && payload.rtfJson) {
+              handleNewValue(payload.rtfJson, payload.locale, payload.rtfHtml);
+            } else {
+              // Fallback for backward compatibility
```

Comment:
Once Storm is deployed it will always send the new payload right? If so then you shouldn't need any fallback.